tests/drop_counters/test_drop_counters.py

```diff
@@ -202,30 +366,35 @@ def base_verification(discard_group, pkt, ptfadapter, duthost, combined_counter,
     testutils.send(ptfadapter, ptf_tx_port_id, pkt, count=PKT_NUMBER)
     time.sleep(1)
 
+
+def str_to_int(value):
+    """ Convert string value which can contain ',' symbols to integer value """
+    return int("".join(value.split(",")))
+
+
+def base_verification(discard_group, pkt, ptfadapter, duthost, ptf_tx_port_id, dut_iface):
+    """
+    Base test function for verification of L2 or L3 packet drops. Verification type depends on 'discard_group' value.
+    Supported 'discard_group' values: 'L2', 'L3', 'ACL'
+    """
+    send_packets(pkt, duthost, ptfadapter, ptf_tx_port_id)
     if discard_group == "L2":
         # Verify drop counter incremented on specific interface
         intf_l2_counters = get_pkt_drops(duthost, GET_L2_COUNTERS)
-        if int(intf_l2_counters[dut_iface][L2_DISCARD_KEY]) != PKT_NUMBER:
+        if int(intf_l2_counters[dut_iface][RX_DRP]) != PKT_NUMBER:
             fail_msg = "'{}' drop counter was not incremented on iface {}. DUT {} == {}; Sent == {}".format(
-                L2_DISCARD_KEY, dut_iface, L2_DISCARD_KEY,
-                int(intf_l2_counters[dut_iface][L2_DISCARD_KEY]), PKT_NUMBER
+                RX_DRP, dut_iface, RX_DRP,
+                int(intf_l2_counters[dut_iface][RX_DRP]), PKT_NUMBER
             )
             pytest.fail(fail_msg)
 
         # Skip L3 discards verification for platform with linked L2 and L3 drop counters
-        if not combined_counter:
-            # Verify other drop counters were not incremented
-            intf_l3_counters = get_pkt_drops(duthost, GET_L3_COUNTERS)
-            unexpected_drops = {}
-            for iface, value in intf_l3_counters.items():
-                if int(value[L3_DISCARD_KEY]) != 0:
-                    unexpected_drops[iface] = int(value[L3_DISCARD_KEY])
-            if unexpected_drops:
-                pytest.fail("L3 'RX_ERR' was incremented for the following interfaces:\n{}".format(unexpected_drops))
+        if not COMBINED_L2L3_DROP_COUNTER:
+            ensure_no_l3_drops(duthost)
     elif discard_group == "L3":
         # Verify L3 drop counter incremented on specific interface
-        l3_drops = get_pkt_drops(duthost, GET_L3_COUNTERS)[dut_iface][L3_DISCARD_KEY]
-        l3_drops = int("".join(l3_drops.split(",")))
+        l3_drops = get_pkt_drops(duthost, GET_L3_COUNTERS)[dut_iface][RX_ERR]
```

Comment:
> I don't understand this change.
> 
> My understanding of devices that combine L2 and L3 drop counts is that all drops will be counted by the same counter, RX_DRP. If that is the case, then any test that specifies a discard group of "L3" will fail for those devices. Did I miss where this is handled?

"My understanding of devices that combine L2 and L3 drop counts is that all drops will be counted by the same counter, RX_DRP" - by L2 RX_DRP.

But for L3 test cases, it sends incorect/error packets, so for L3 test cases will be incremented L3 RX_ERR.

So logic of L3 test cases will be verified.

Reply:
I think it still ends up in RX_DRP in the L3 case. My experience with the combined dropped counters is they all end up in the same counter.

Reply:
To check L2/L3 drop counters are used different commands.
In L3 case **there is no "RX_DRP" field.**

To check drops on L2 level it is used the following command:
"portstat -j"
Its output:
     "Ethernet0": {
        "RX_BPS": "0.00 B/s",
        "RX_DRP": "0",
        "RX_ERR": "0",
        "RX_OK": "0",
        "RX_OVR": "N/A",
        "RX_UTIL": "0.00%",
        "STATE": "U",
        "TX_BPS": "7.08 B/s",
        "TX_DRP": "0",
        "TX_ERR": "0",
        "TX_OK": "15",
        "TX_OVR": "N/A",
        "TX_UTIL": "0.00%"
    }

To check drops on L3 level it is used the following command:
    "Ethernet0": {
        "RX_BPS": "0.00 B/s",
        "RX_ERR": "0",
        "RX_OK": "0",
        "RX_PPS": "0.00/s",
        "TX_BPS": "0.00 B/s",
        "TX_ERR": "0",
        "TX_OK": "0",
        "TX_PPS": "0.00/s"
    }

So counters are checked correctly.

Reply:
I think I misunderstood something along the way then. Devices with the combined counters should be using the same command and the same field for L2 and L3. If I try to run intfstat on a Broadcom device, for instance, I get no output. For such devices, these tests will not work as intended.

Reply:
Maybe now I don't understand you. So again:
- Devices that combine L2 and L3 drop counts all drops by the same counter - RX_DRP, by calling CLI command "portstat -j".

- Devices that do not combine L2 and L3 drop counts separately:
L2 drop - RX_DRP, by calling CLI command "portstat -j"
L3 drop - RX_ERR, by calling CLI command "intfstat -j"

So those commands are also used by test case.

Is that correct behavior or not?

Reply:
That's the correct behavior. I am just confused where this case is being handled:
```
Devices that combine L2 and L3 drop counts all drops by the same counter - RX_DRP, by calling CLI command "portstat -j".
```

When `discard_group` is L3. From what I can see it just looks like `intfstat` is being used whenever `discard_group` is L3.

Reply:
Now its clear, good catch, thank you :)
Can you please take a look now?

Reply:
There we go, that looks good now! :) One little thing, I'm not sure how `ensure_no_l3_drops` will handle the CLI printing out something like "N/A", so it might be best to skip that check for the combined case.

Reply:
Also, should we include a similar if-case for the ACL drops case as well?

Reply:
Fixed `ensure_no_l3_drops` and `ensure_no_l2_drops` to print warning in case of not integer value occurred. So it will give possibility still to check that no redundant drops occurred in normal case.

Reply:
Sounds good to me! :)